pil-analyzer/src/condenser.rs

```diff
@@ -596,9 +602,38 @@ impl<'a, T: FieldElement> SymbolLookup<'a, T> for Condenser<'a, T> {
 
                 self.new_prover_functions.push(e);
             }
-            _ => self
-                .new_constraints
-                .push(to_constraint(&constraints, source, &mut self.counters)),
+            _ => self.new_constraints.push((constraints, source)),
+        }
+        Ok(())
+    }
+
+    fn capture_constraints(
+        &mut self,
+        fun: Arc<Value<'a, T>>,
+    ) -> Result<Arc<Value<'a, T>>, EvalError> {
+        let existing_constraints = self.new_constraints.len();
+        let result = evaluate_function_call(fun, vec![], self);
+        let constrs = self
+            .new_constraints
+            .drain(existing_constraints..)
+            .map(|(c, _)| c)
+            .collect();
+        // TODO we could now subtract constrs.len() from the identity counter.
```

Comment:
Is there a reason not to do it in this PR?

Reply:
I'm not sure if it's correct :)

Reply:
Ah no, it's even better: Since I changed the type of the new stored constraints to Value instead of Identity, the identity IDs are only assigned in the `extract_new_constraits` call, so we don't need to reset IDs because they have not been assigned yet.